src/stash/src/memory.rs

```diff
@@ -7,94 +7,171 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use std::collections::hash_map::Entry;
```

Comment:
This file is the "new" implementation, so the diff is a bit misleading, and this is probably better read without a diff. Also, maybe doesn't need much scrutiny since it's only used in tests.